Gores/WaterDroplet/CragsLavaDroplet.cs

```diff
@@ -0,0 +1,157 @@
+                {
+                    frameDuration = 24 + Main.rand.Next(256);
+                }
+
+                if (gore.frame == 3)
+                {
+                    frameDuration = 24 + Main.rand.Next(96);
+                }
+
+                if (gore.frameCounter >= frameDuration)
+                {
+                    gore.frameCounter = 0;
+                    gore.frame += 1;
+                    if (gore.frame == 5)
+                    {
+                        int droplet = Gore.NewGore(new EntitySource_Misc(nameof(CragsLavaDroplet)), gore.position, gore.velocity, gore.type);
+                        Main.gore[droplet].frame = 9;
+                        Main.gore[droplet].velocity *= 0f;
+                    }
+                }
+            }
+            else if (gore.frame <= 6)
+            {
+                frameDuration = 8;
+                if (gore.frameCounter >= frameDuration)
+                {
+                    gore.frameCounter = 0;
+                    gore.frame += 1;
+                    if (gore.frame == 7)
+                    {
+                        gore.active = false;
+                    }
+                }
+            }
+            else if (gore.frame <= 9)
+            {
+                frameDuration = 6;
+                gore.velocity.Y += 0.2f;
+                if (gore.velocity.Y < 0.5f)
+                {
+                    gore.velocity.Y = 0.5f;
+                }
+
+                if (gore.velocity.Y > 12f)
+                {
+                    gore.velocity.Y = 12f;
+                }
+
+                if (gore.frameCounter >= frameDuration)
+                {
+                    gore.frameCounter = 0;
+                    gore.frame += 1;
+                }
+
+                if (gore.frame > 9)
+                {
+                    gore.frame = 7;
+                }
+            }
+            else
+            {
+                gore.velocity.Y += 0.1f;
+                if (gore.frameCounter >= frameDuration)
+                {
+                    gore.frameCounter = 0;
+                    gore.frame += 1;
+                }
+
+                gore.velocity *= 0f;
+                if (gore.frame > 14)
+                {
+                    gore.active = false;
+                }
+            }
+			float num24 = 1f;
+			float num25 = 1f;
+			float num26 = 1f;
+			float num27 = 0.6f;
+			num27 = ((gore.frame == 0) ? (num27 * 0.1f) : ((gore.frame == 1) ? (num27 * 0.2f) : ((gore.frame == 2) ? (num27 * 0.3f) : ((gore.frame == 3) ? (num27 * 0.4f) : ((gore.frame == 4) ? (num27 * 0.5f) : ((gore.frame == 5) ? (num27 * 0.4f) : ((gore.frame == 6) ? (num27 * 0.2f) : ((gore.frame <= 9) ? (num27 * 0.5f) : ((gore.frame == 10) ? (num27 * 0.5f) : ((gore.frame == 11) ? (num27 * 0.4f) : ((gore.frame == 12) ? (num27 * 0.3f) : ((gore.frame == 13) ? (num27 * 0.2f) : ((gore.frame != 14) ? 0f : (num27 * 0.1f))))))))))))));
```

Comment:
I'm not a official dev or anything. But I'd like to suggest C# 8.0's simplified switch statement for this if you don't mind (Unless it's some kind of meme)

Like:
```cs
num27 *= goreframe switch
{
    0 => 0.1f,
    1 => 0.2f,
    2 => 0.3f,
    // ...
    _ => 0.0f, //default case
};
```

Reply:
yeah I'll change this, its decompiled vanilla code so thats why is so _freaky_

Reply:
Changed this to be a switch instead of this abomination, thanks for the suggestion flowaria